### PR TITLE
Clean up `add_to_list` function a bit.

### PR DESCRIPTION
**Pull request by iphydf**
Can't trivially get rid of recursion here, since it's a non-linear
recursive function.

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/1140)

### CODE REVIEW

File: toxcore/DHT.c [hunk 1]
```diff
@@ -755,30 +755,26 @@ static int client_or_ip_port_in_list(const Logger *log, const Mono_Time *mono_ti
...
                 add_to_list(nodes_list, length, pk_bak, ip_port_bak, cmp_pk);
             }
 
-            return 1;
+            return true;
         }
     }
 
-    return 0;
+    return true;
```
Comment (zugz) [inline toxcore/DHT.c]: false

File: toxcore/DHT.h [hunk 2]
```diff
@@ -304,7 +304,7 @@ int id_closest(const uint8_t *pk, const uint8_t *pk1, const uint8_t *pk2);
 
 /* Add node to the node list making sure only the nodes closest to cmp_pk are in the list.
  */
```
Comment (zugz) [inline toxcore/DHT.h]: Explain return value

File: toxcore/DHT.c (hunk 1 above)
Comment (iphydf) [reply toxcore/DHT.c]: Done.

File: toxcore/DHT.h (hunk 2 above)
Comment (iphydf) [reply toxcore/DHT.h]: Done.

**Pull request merged**